controllers/datafoundationclient.go

```diff
@@ -208,3 +234,44 @@ func (r *dataFoundationClientReconciler) reconcileDefaultFSStorageClassClaim() e
 
 	return nil
 }
+
+func (r *dataFoundationClientReconciler) hasOCSVolumes() (bool, error) {
+	storageClassList := &storagev1.StorageClassList{}
+	if err := r.list(storageClassList); err != nil {
+		return false, fmt.Errorf("failed to list Storage Classes: %v", err)
+	}
+	ocsStorageClass := make(map[string]bool)
+	for i := range storageClassList.Items {
+		storageClassName := storageClassList.Items[i].Name
+		if storageClassName == defaultBlockStorageClassClaimName || storageClassName == defaultFSStorageClassClaimName {
+			ocsStorageClass[storageClassName] = true
+		}
+	}
```

Comment:
This entire loop does not make sense. In the end, you will get a map that only contains the following pairs:
```
defaultBlockStorageClassClaimName -> true
defaultFSStorageClassClaimName -> true
```

It is basically equivalent to 
```
ocsStorageClass[defaultBlockStorageClassClaimName] = true
ocsStorageClass[defaultFSStorageClassClaimName] = true
```

But there is an even larger issue here, it does NOT take into account any additional `storageclass`/`storageclassclaims` that the customer created (which are not the default ones)

The condition needs to be based on the Provisioner field of the storageclass not on the name.

Reply:
Fixed by checking provisioner name instead